pkg/controllers/subnet/subnet_webhook.go

```diff
@@ -10,23 +10,35 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
 
 	"github.com/vmware-tanzu/nsx-operator/pkg/apis/vpc/v1alpha1"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
 	"github.com/vmware-tanzu/nsx-operator/pkg/util"
 )
 
 var NSXOperatorSA = "system:serviceaccount:vmware-system-nsx:ncp-svc-account"
 
-// Create validator instead of using the existing one in controller-runtime because the existing one can't
+// Create a validator instead of using the existing one in controller-runtime because the existing one can't
 // inspect admission.Request in Handle function.
 
-// +kubebuilder:webhook:path=/validate-crd-nsx-vmware-com-v1alpha1-subnet,mutating=false,failurePolicy=fail,sideEffects=None,groups=crd.nsx.vmware.com,resources=subnets,verbs=delete,versions=v1alpha1,name=subnet.validating.crd.nsx.vmware.com,admissionReviewVersions=v1
+// +kubebuilder:webhook:path=/validate-crd-nsx-vmware-com-v1alpha1-subnet,mutating=false,failurePolicy=fail,sideEffects=None,groups=crd.nsx.vmware.com,resources=subnets,verbs=create;update;delete,versions=v1alpha1,name=subnet.validating.crd.nsx.vmware.com,admissionReviewVersions=v1
 
 type SubnetValidator struct {
 	Client  client.Client
 	decoder admission.Decoder
 }
 
+// isSharedSubnet checks if a Subnet is shared based on the associated_resource annotation
+func isSharedSubnet(subnet *v1alpha1.Subnet) bool {
+	if subnet.Annotations != nil {
+		if _, ok := subnet.Annotations[common.AnnotationAssociatedResource]; ok {
+			return true
+		}
+	}
+	return subnet.Status.Shared
```

Comment:
If there is a case that shared is true while the Subnet CR does not have an annotation? Or should we just return false here?

Reply:
fixed